cluster-autoscaler/estimator/estimator.go

```diff
@@ -38,14 +38,16 @@ type Estimator interface {
 }
 
 // EstimatorBuilder creates a new estimator object.
-type EstimatorBuilder func(*simulator.PredicateChecker) Estimator
+type EstimatorBuilder func(simulator.PredicateChecker, simulator.ClusterSnapshot) Estimator
```

Comment:
nit: I think it would be more elegant to pass snapshot to Estimate(), rather than keep a single snapshot as part of state and mutate it. This can be done as a follow-up if you agree, but let's not refactor in this PR.

Reply:
I will put this to as maybe_todo in the follwup list for now